src/catalog/src/durable/upgrade/persist/v47_to_v48.rs

```diff
@@ -0,0 +1,790 @@
+                },
+            )),
+        };
+        let new = v48::StateUpdateKind {
+            kind: Some(v48::state_update_kind::Kind::Database(
+                v48::state_update_kind::Database {
+                    key: Some(new_key),
+                    value: Some(new_value),
+                },
+            )),
+        };
+        let action = MigrationAction::Update(old, new);
+        migrations.push(action);
+    }
+
+    let mut system_schemas: BTreeMap<u64, u32> = [
+        // mz_catalog
+        (1, 16_656),
+        // pg_catalog
+        (2, 16_657),
+        // mz_internal
+        (4, 16_658),
+        // information_schema
+        (5, 16_659),
+        // mz_unsafe
+        (6, 16_660),
+    ]
+    .into_iter()
+    .collect();
+    for schema in schemas {
+        let old_key = schema.key.expect("missing key field");
+        let old_value = schema.value.expect("missing value field");
+        let new_key = WireCompatible::convert(&old_key);
+        let oid = match old_key
+            .id
+            .as_ref()
+            .expect("missing id field")
+            .value
+            .as_ref()
+            .expect("missing value field")
+        {
+            v47::schema_id::Value::System(id) => {
+                system_schemas.remove(id).expect("unexpected system schema")
+            }
+            v47::schema_id::Value::User(_) => {
+                let oid = cur_user_oid;
+                cur_user_oid += 1;
+                oid
+            }
+        };
+        let new_value = v48::SchemaValue {
+            database_id: old_value.database_id.as_ref().map(WireCompatible::convert),
+            name: old_value.name.clone(),
+            owner_id: old_value.owner_id.as_ref().map(WireCompatible::convert),
+            privileges: old_value
+                .privileges
+                .iter()
+                .map(WireCompatible::convert)
+                .collect(),
+            oid,
+        };
+        let old = v47::StateUpdateKind {
+            kind: Some(v47::state_update_kind::Kind::Schema(
+                v47::state_update_kind::Schema {
+                    key: Some(old_key),
+                    value: Some(old_value),
+                },
+            )),
+        };
+        let new = v48::StateUpdateKind {
+            kind: Some(v48::state_update_kind::Kind::Schema(
+                v48::state_update_kind::Schema {
+                    key: Some(new_key),
+                    value: Some(new_value),
+                },
+            )),
+        };
+        let action = MigrationAction::Update(old, new);
+        migrations.push(action);
+    }
+
+    let mut system_roles: BTreeMap<u64, u32> = [
+        // mz_system
+        (1, 16_661),
+        // mz_support
+        (2, 16_662),
+        // mz_monitor
+        (3, 16_663),
+        // mz_monitor_redacted
+        (4, 16_664),
+    ]
+    .into_iter()
+    .collect();
+    let public_role_oid: u32 = 16_944;
```

Comment:
small preference to move these to consts at the top of the file, up to you though

Reply:
Good idea, updated.